compiler/rustc_middle/src/ty/consts/valtree.rs

```diff
@@ -104,6 +104,15 @@ impl<'tcx> ValTree<'tcx> {
                 }
                 _ => {}
             },
+            ty::Array(array_ty, _) if *array_ty == tcx.types.u8 => {
+                let leafs = self
+                    .unwrap_branch()
+                    .into_iter()
+                    .map(|v| v.unwrap_leaf().try_to_u8().unwrap())
+                    .collect::<Vec<_>>();
+
+                return Some(tcx.arena.alloc_from_iter(leafs.into_iter()));
```

Comment:
is it possible to not collect as an intermediate step but just pass the iterator directly to the arena?